app/ide-desktop/lib/content/src/newtype.ts

```diff
@@ -0,0 +1,37 @@
+/** @file TypeScript's closest equivalent of `newtype`s. */
+
+interface Brand<TypeName extends string> {
```

Comment:
I don't think that introducing the name `Brand` next to `Newtype` is a good idea. It just makes understanding it harder. What about using `NewtypeVariant` instead? Or `newtype.Variant`. Newtype variants are again, widely used and understood, so probably this will be slightly better name. What do you think ? :)

Reply:
that's a good idea, i wasn't aware of that term